pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -2704,8 +2709,10 @@ func (p *LogicalExpand) ExhaustPhysicalPlans(prop *property.PhysicalProperty) ([
 	if prop.TaskTp == property.MppTaskType && prop.MPPPartitionTp != property.AnyType {
 		return nil, true, nil
 	}
+	var physicalExpands []base.PhysicalPlan
 	// for property.RootTaskType and property.MppTaskType with no partition option, we can give an MPP Expand.
-	if p.SCtx().GetSessionVars().IsMPPAllowed() {
+	canPushToTiFlash := p.CanPushToCop(kv.TiFlash)
```

Comment:
"CanPushToCop" This function name is strange. Cop is a protocol name but TiFlash already deprecated TiFlashCop and only keep the TiFlash[MPP]

Reply:
yes, borrowed from old logic, i think it should be renamed as p.CanPushToEngine, since your arguments is engine type, the name will be more accurate.